### ISSUE TITLE
Notice: get_product is deprecated and Notice: post was called incorrectly and  Notice: WC_Product::get_gallery_attachment_ids is deprecated  appear on front-end of product page

### ISSUE THREAD

**Issue by Pcosta88** [Yoast/wpseo-woocommerce#154]
### What did you expect to happen?
No notices would appear.
### What happened instead?
They did
### How can we reproduce this behavior?
**1.Create a product**
**2. Publish it**
**3.Activate Woocommerce SEO**
**4.See warnings on Front-end**

![screen shot 2017-05-17 at 3 42 45 pm](https://cloud.githubusercontent.com/assets/8238074/26172758/7edf1c4a-3b17-11e7-80a6-92842f893250.png)

```
Notice: get_product is deprecated since version 3.0! Use wc_get_product instead. in /Applications/mampstack-7.0.12-0/apps/wordpress/htdocs/wp-includes/functions.php on line 3830

Notice: post was called incorrectly. Product properties should not be accessed directly. Backtrace: require('wp-blog-header.php'), require_once('wp-includes/template-loader.php'), include('/plugins/woocommerce/templates/single-product.php'), get_header, locate_template, load_template, require_once('/themes/storefront/header.php'), wp_head, do_action('wp_head'), WP_Hook->do_action, WP_Hook->apply_filters, call_user_func_array, WPSEO_Frontend->head, do_action('wpseo_head'), WP_Hook->do_action, WP_Hook->apply_filters, call_user_func_array, WPSEO_Frontend->metadesc, WPSEO_Frontend->generate_metadesc, apply_filters('wpseo_metadesc'), WP_Hook->apply_filters, call_user_func_array, Yoast_WooCommerce_SEO->metadesc, WC_Abstract_Legacy_Product->__get, wc_doing_it_wrong Please see Debugging in WordPress for more information. (This message was added in version 3.0.) in /Applications/mampstack-7.0.12-0/apps/wordpress/htdocs/wp-includes/functions.php on line 4138

Notice: post was called incorrectly. Product properties should not be accessed directly. Backtrace: require('wp-blog-header.php'), require_once('wp-includes/template-loader.php'), include('/plugins/woocommerce/templates/single-product.php'), get_header, locate_template, load_template, require_once('/themes/storefront/header.php'), wp_head, do_action('wp_head'), WP_Hook->do_action, WP_Hook->apply_filters, call_user_func_array, WPSEO_Frontend->head, do_action('wpseo_head'), WP_Hook->do_action, WP_Hook->apply_filters, call_user_func_array, WPSEO_Frontend->metadesc, WPSEO_Frontend->generate_metadesc, apply_filters('wpseo_metadesc'), WP_Hook->apply_filters, call_user_func_array, Yoast_WooCommerce_SEO->metadesc, WC_Abstract_Legacy_Product->__get, wc_doing_it_wrong Please see Debugging in WordPress for more information. (This message was added in version 3.0.) in /Applications/mampstack-7.0.12-0/apps/wordpress/htdocs/wp-includes/functions.php on line 4138

Notice: WC_Product::get_gallery_attachment_ids is deprecated since version 3.0! Use WC_Product::get_gallery_image_ids instead. in /Applications/mampstack-7.0.12-0/apps/wordpress/htdocs/wp-includes/functions.php on line 3830

```

**5.Deactivate WooSEO and see errors resolve**

![screen shot 2017-05-17 at 3 44 08 pm](https://cloud.githubusercontent.com/assets/8238074/26172833/b46e4fd4-3b17-11e7-990e-34c96a0e406b.png)

Can you provide a link to a page which shows this issue?

### Technical info
* WordPress version:4.7.5
* Yoast SEO version: 4.7.1
* WooCommerce SEO: 4.7
* WooCommerce: 3.0.7 (tested with this one but user reports it happening on 3.0.6)

**Comment by Pcosta88** [comment]
Please inform the customer of conversation # 196580 when this conversation has been closed.

**Comment by Pcosta88** [comment]
Please inform the customer of conversation # 195267 when this conversation has been closed.

**Comment by benvaassen** [comment]
This only happens when debugging is on: `define( 'WP_DEBUG', true );`

**Comment by suascat** [comment]
Please inform the customer of conversation # 799185 when this conversation has been closed.

**Comment by CarolineGeven** [comment]
This issue has been fixed in version 4.9 which is scheduled for release this week.
